packages/api/tests/endpoints/test-rules.js

```diff
@@ -305,6 +293,89 @@ test('POST returns a record exists when one exists', async (t) => {
   t.falsy(record);
 });
 
+test('POST returns a 400 response if record is missing a required property', async (t) => {
+  const newRule = fakeRuleFactoryV2();
+
+  // Remove required property to trigger create error
+  delete newRule.collection;
+
+  const response = await request(app)
+    .post('/rules')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(newRule)
+    .expect(400);
+  t.is(response.status, 400);
+});
+
+test('POST returns a 400 response if rule name is invalid', async (t) => {
+  const newRule = fakeRuleFactoryV2({
+    name: 'bad rule name'
+  });
+
+  const response = await request(app)
+    .post('/rules')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(newRule)
+    .expect(400);
+  t.is(response.status, 400);
+});
+
+test('POST returns a 400 response if rule type is invalid', async (t) => {
+  const newRule = fakeRuleFactoryV2({
+    type: 'invalid'
+  });
+
+  const response = await request(app)
+    .post('/rules')
+    .set('Accept', 'application/json')
+    .set('Authorization', `Bearer ${jwtAuthToken}`)
+    .send(newRule)
+    .expect(400);
+  t.is(response.status, 400);
+});
+
+test.serial('POST returns a 500 response if workflow definition file does not exist', async (t) => {
```

Comment:
I debated whether this should be a 400 error instead of 500. But ultimately a workflow definition file could be missing because of a deployment failure (or accidentally deleting files from S3) even if the request is perfectly valid, so 500 felt like the appropriate response